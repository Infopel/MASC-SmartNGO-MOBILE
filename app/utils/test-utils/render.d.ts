export type RenderOptions = {
  wrapper?: React.ComponentType<any>
  createNodeMock?: (element: React.ReactElement) => any
}
export type AllTheProvidersProps = { children: React.ReactNode }
