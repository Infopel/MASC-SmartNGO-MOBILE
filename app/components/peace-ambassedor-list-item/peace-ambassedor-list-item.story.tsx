import * as React from "react"
import { storiesOf } from "@storybook/react-native"
import { StoryScreen, Story, UseCase } from "../../../storybook/views"
import { color } from "../../theme"
import { PeaceAmbassedorListItem } from "./peace-ambassedor-list-item.component"

storiesOf("PeaceAmbassedorListItem", module)
  .addDecorator((fn) => <StoryScreen>{fn()}</StoryScreen>)
  .add("Style Presets", () => (
    <Story>
      <UseCase text="Primary" usage="The primary.">
        <PeaceAmbassedorListItem style={{ backgroundColor: color.error }} />
      </UseCase>
    </Story>
  ))
