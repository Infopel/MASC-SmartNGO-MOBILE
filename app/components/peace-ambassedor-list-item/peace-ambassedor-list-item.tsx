import { Card } from "components/card/card"
import { useIsPressed } from "components/hooks"
import { Text } from "components/text/text"
import { Avatar, HStack, Pressable, usePropsResolution, VStack } from "native-base"
import * as React from "react"
import { StyleProp, ViewStyle } from "react-native"
import { IPeaceEmbassadorVDO } from "storage/queries/initiative"
import { DateUtils } from "utils/DateUitls"

export interface PeaceAmbassedorListItemProps {
  /**
   * An optional style override useful for padding & margin.
   */
  style?: StyleProp<ViewStyle>

  item: IPeaceEmbassadorVDO

  onNavigate?: () => void
}

/**
 * Describe your component here
 */
export const PeaceAmbassedorListItem = function PeaceAmbassedorListItem(props: PeaceAmbassedorListItemProps) {
  const { style } = props
  const { onNavigate, item, ...rest } = props

  const { isPressed, pressableProps } = useIsPressed()

  const { ...resolvedNavigateProps } = usePropsResolution('Button', { onPress: onNavigate, variant: 'ghost', ...rest, }, { isPressed, isDisabled: onNavigate === undefined })

  return (
    <Card borderLeftWidth={6} borderLeftColor='primary.500'>
      <Pressable accessibilityRole="button"  {...resolvedNavigateProps} {...pressableProps} >
        <HStack>
          <VStack flex='1' space={2}>
            <Text fontWeight={'bold'}>{item.name}</Text>
            <Text>{(item.province ? item.province + ", " : "") + item.district}</Text>
            <Text>{DateUtils.formatToShortDate(item.date)}</Text>
          </VStack>
          <Avatar >{item.numberOfActiveMembers}</Avatar>
        </HStack>
      </Pressable>
    </Card>
  )
}