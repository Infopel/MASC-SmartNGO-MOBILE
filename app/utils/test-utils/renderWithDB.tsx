jest.mock("storage/adapter.ts")
import DatabaseProvider from "@nozbe/watermelondb/DatabaseProvider"
import { render } from "@testing-library/react-native"
import { default as React } from "react"
import { database } from "storage/database"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProvidersWithDB({ children }: AllTheProvidersProps) {
    return (
        <DatabaseProvider database={database}>
            <ThemeProvider>{children}</ThemeProvider>
        </DatabaseProvider>
    )
}

export function renderWithDB<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
    return render(ui, { wrapper: AllTheProvidersWithDB, ...options })
}


export class DiagnosticError extends Error {
    name: string = "Diagnostic error"
  }
  