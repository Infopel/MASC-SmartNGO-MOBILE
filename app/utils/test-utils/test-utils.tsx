import { render } from "@testing-library/react-native"
import React from "react"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProviders({ children }: AllTheProvidersProps) {
  return (
    <ThemeProvider>{children}</ThemeProvider>
  )
}

function customRender<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

// re-export everything
export * from "@testing-library/react-native"
// override render method
export { customRender as render }

