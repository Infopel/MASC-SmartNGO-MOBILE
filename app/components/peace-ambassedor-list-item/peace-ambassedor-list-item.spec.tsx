import { fireEvent, render } from "utils/test-utils/test-utils"
import * as React from "react"
import { PeaceAmbassedorListItem } from "./peace-ambassedor-list-item"



describe("<PeaceAmbassedorListItem>", () => {
  it("render", async function () {
    const component = <PeaceAmbassedorListItem item={{
      date: 3213213321, district: "Moamba", id: "asdasdsadsad", name: "Judas",
    }}></PeaceAmbassedorListItem>

    expect(render(component).toJSON()).toMatchSnapshot()
  })
})
