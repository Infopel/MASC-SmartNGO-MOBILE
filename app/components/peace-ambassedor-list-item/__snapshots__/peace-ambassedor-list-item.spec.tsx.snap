// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PeaceAmbassedorListItem> render 1`] = `
<View
  dataSet={Object {}}
  style={
    Object {
      "backgroundColor": "#fafaf9",
      "borderBottomColor": "#fafafa",
      "borderLeftColor": "#ef8812",
      "borderLeftWidth": 6,
      "borderRadius": 6,
      "borderRightColor": "#fafafa",
      "borderTopColor": "#fafafa",
      "borderWidth": 1,
      "elevation": 2,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 1.41,
    }
  }
>
  <View
    _icon={
      Object {
        "color": "primary.600",
        "size": "sm",
      }
    }
    _spinner={
      Object {
        "color": "primary.600",
        "focusable": false,
        "size": "sm",
      }
    }
    _stack={
      Object {
        "alignItems": "center",
        "space": "1.5",
      }
    }
    _text={
      Object {
        "color": "primary.600",
        "fontSize": "sm",
      }
    }
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    colorScheme="primary"
    dataSet={Object {}}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 4,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 0.4,
        "paddingBottom": 10,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 10,
      }
    }
  >
    <View
      dataSet={Object {}}
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        dataSet={Object {}}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          dataSet={Object {}}
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#171717",
              "fontFamily": undefined,
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "700",
              "letterSpacing": 0,
              "lineHeight": 21,
              "textDecorationLine": undefined,
            }
          }
        >
          Judas
        </Text>
        <View
          dataSet={Object {}}
          style={
            Object {
              "height": 8,
            }
          }
        />
        <Text
          dataSet={Object {}}
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#171717",
              "fontFamily": undefined,
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 21,
              "textDecorationLine": undefined,
            }
          }
        >
          Moamba
        </Text>
        <View
          dataSet={Object {}}
          style={
            Object {
              "height": 8,
            }
          }
        />
        <Text
          dataSet={Object {}}
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#171717",
              "fontFamily": undefined,
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 21,
              "textDecorationLine": undefined,
            }
          }
        >
          07/02/1970
        </Text>
      </View>
      <View
        dataSet={Object {}}
        style={Object {}}
      />
      <View
        dataSet={Object {}}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#a1a1aa",
            "borderColor": "#27272a",
            "borderRadius": 9999,
            "height": 48,
            "justifyContent": "center",
            "position": "relative",
            "width": 48,
          }
        }
      />
    </View>
  </View>
</View>
`;
